code/modules/logging/bb.dm

```diff
@@ -0,0 +1,291 @@
+/// Convert almost anything to a JSON-safe value.
+///
+/// Inputs **must not** be self-referential.
+/// Associative lists with non-{number, string} keys are discarded.
+/proc/_bb_jsonify(data as anything)
+	if(islist(data))
+		. = list()
+
+		for(var/key in data)
+			var/value
+
+			try
+				value = data[key]
+			catch
+				// Expected, indicates a flat list
+
+			if(!isnull(value))
+				if(!istext(key) && !isnum(key))
+					// We can't store non-numeric/non-string keys in JSON maps.
+					// Skip this one.
+					continue
+
+				.[key] = _bb_jsonify(data[key])
+			else if(!isnull(key))
+				. += list(_bb_jsonify(key))
```

Comment:
This isn't actually safe as an assumption.
byond doesn't distinguish between an array and a map, they are the same thing and can be intermixed.
all this try/catch block will catch is out of bouds accesses, but if we have a list like this
`list(2, "jerry I'm a flat list", "bill I know you told me yesterday")`
This code would runtime on line 23 (due to an index out of bounds exception)